Userland/Applications/Assistant/Providers.cpp

```diff
@@ -75,4 +91,16 @@ void CalculatorProvider::query(String const& query, Function<void(Vector<Nonnull
     on_complete(results);
 }
 
+void TerminalProvider::query(String const& query, Function<void(Vector<NonnullRefPtr<Result>>)> on_complete)
+{
+    if (!query.starts_with('$'))
+        return;
+
+    auto command = query.substring(1);
+
+    Vector<NonnullRefPtr<Result>> results;
```

Comment:
You could use `NonnullRefPtrVector` here to make it a bit cleaner, but this would require changing all the other code, so you don't need to change it.

Reply:
Yeahhh I thought about making a separate commit to change all of these, but moved onto something else :) this one was just copy-pasted from another provider. I'll go ahead and make that commit anyways.